pipeline/post-results-new.groovy

```diff
@@ -0,0 +1,278 @@
+// Pipeline script for uploading IBM test run results to report portal.
+def credsRpProc = [:]
+def sharedLib
+def rpPreprocDir
+def tierLevel = null
+def stageLevel = null
+def run_type = "Sanity Run"
+def build_url
+def reportBucket = "qe-ci-reports"
+def remoteName= "ibm-cos"
+def msgMap = [:]
+def tags_list
+def umbLib
+def composeInfo
+def metaData = [:]
+def rp_base_link = "https://reportportal-rhcephqe.apps.ocp-c1.prod.psi.redhat.com"
+
+node("rhel-8-medium || ceph-qe-ci") {
+
+    try {
+        stage('prepareJenkinsAgent') {
+            if (env.WORKSPACE) { sh script: "sudo rm -rf * .venv" }
+            checkout(
+                scm: [
+                    $class: 'GitSCM',
+                    branches: [[name: "origin/master"]],
+                    extensions: [[
+                        $class: 'CleanBeforeCheckout',
+                        deleteUntrackedNestedRepositories: true
+                    ], [
+                        $class: 'WipeWorkspace'
+                    ], [
+                        $class: 'CloneOption',
+                        depth: 1,
+                        noTags: true,
+                        shallow: true,
+                        timeout: 10,
+                        reference: ''
+                    ]],
+                    userRemoteConfigs: [[
+                        url: "https://github.com/red-hat-storage/cephci.git"
+                    ]]
+                ],
+                changelog: false,
+                poll: false
+            )
+
+            // prepare the node
+            sharedLib = load("${env.WORKSPACE}/pipeline/vars/v3.groovy")
+            sharedLib.prepareNode(3)
+
+            msgMap = sharedLib.getCIMessageMap()
+            println("msgMap : ${msgMap}")
+            println("sharedLib: ${sharedLib}")
+
+            emailLib = load("${env.WORKSPACE}/pipeline/vars/email.groovy")
+            println("emailLib : ${emailLib}")
+            reportLib = load("${env.WORKSPACE}/pipeline/vars/upload_results.groovy")
+            println("reportLib : ${reportLib}")
+        }
+
+        stage('updatePipelineMetadata'){
+            if ( msgMap["pipeline"].containsKey("tags") ) {
+                def tag = msgMap["pipeline"]["tags"]
+                println("Tag : ${tag}")
+                tags_list = tag.split(',') as List
+                def stage_index = tags_list.findIndexOf { it ==~ /stage-\w+/ }
+                stageLevel = tags_list.get(stage_index)
+                println("Stage : ${stageLevel}")
+                def tier_index = tags_list.findIndexOf { it ==~ /tier-\w+/ }
+                tierLevel = tags_list.get(tier_index)
+                println("Tier : ${tierLevel}")
+                run_type = msgMap["pipeline"]["run_type"]
+                println("Type : ${run_type}")
+                build_url = msgMap["run"]["url"]
+                println("Build : ${build_url}")
+            }
+        }
+
+        stage('configureReportPortalWorkDir') {
```

Comment:
Instead of separate stage, make this part of `prepareJenkinsAgent`

Reply:
This is a logically separate action being done, so i felt keeping it as a separate stage made sense.